### ISSUE TITLE
Support Python 3.13

### ISSUE THREAD

**Issue by edgarrmondragon** [jelmer/dulwich#1352]
Python 3.13.0rc1 is out[^1] so it'd be nice to start publishing the corresponding wheels for this package.

[^1]: https://discuss.python.org/t/python-3-13-0-release-candidate-1-released/59703

**Comment by jelmer** [comment]
It should just be a matter of updating the files in .github/workflows/ to include 3.13. can you submit a PR?

**Comment by edgarrmondragon** [comment]
> can you submit a PR?

Yes, will do!

**Comment by jelmer** [comment]
Reopening this since the actual tests are failing. This appears to be blocked on https://github.com/PyO3/pyo3/pull/4420

**Comment by jelmer** [comment]
We're also hitting https://github.com/python/cpython/issues/122728:

```
ERROR: test_default_key (tests.test_porcelain.CommitSignTests.test_default_key)
----------------------------------------------------------------------
SystemError: Objects/dictobject.c:3774: bad argument to internal function

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/runner/work/dulwich/dulwich/tests/test_porcelain.py", line 469, in test_default_key
    sha = porcelain.commit(
        self.repo.path,
    ...<3 lines>...
        signoff=True,
    )
  File "/home/runner/work/dulwich/dulwich/dulwich/porcelain.py", line 444, in commit
    return r.do_commit(
           ~~~~~~~~~~~^
        message=message,
        ^^^^^^^^^^^^^^^^
    ...<6 lines>...
        sign=signoff if isinstance(signoff, (str, bool)) else None,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/home/runner/work/dulwich/dulwich/dulwich/repo.py", line 1045, in do_commit
    c.sign(keyid)
    ~~~~~~^^^^^^^
  File "/home/runner/work/dulwich/dulwich/dulwich/objects.py", line 1508, in sign
    self.gpgsig, unused_result = c.sign(
                                 ~~~~~~^
        self.as_raw_string(), mode=gpg.constants.sig.mode.DETACH
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/hostedtoolcache/Python/3.13.0-rc.1/x64/lib/python3.13/site-packages/gpg/core.py", line 444, in sign
    result = self.op_sign_result()
  File "/opt/hostedtoolcache/Python/3.13.0-rc.1/x64/lib/python3.13/site-packages/gpg/core.py", line 151, in wrapper
    return _funcwrap(self, *args)
  File "/opt/hostedtoolcache/Python/3.13.0-rc.1/x64/lib/python3.13/site-packages/gpg/core.py", line 138, in _funcwrap
    result = func(slf.wrapped, *args)
  File "/opt/hostedtoolcache/Python/3.13.0-rc.1/x64/lib/python3.13/site-packages/gpg/gpgme.py", line 1144, in gpgme_op_sign_result
    return _gpgme.gpgme_op_sign_result(ctx)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^
  File "<frozen importlib._bootstrap>", line 645, in parent
SystemError: <sys.legacy_event_handler object at 0x7f3258f7c230> returned a result with an exception set
```

**Comment by skshetry** [comment]
@jelmer, do you plan to make a release? I am interested in 3.13 support and the fix from https://github.com/jelmer/dulwich/pull/1300.

**Comment by jelmer** [comment]
> @jelmer, do you plan to make a release? I am interested in 3.13 support and the fix from #1300.

I'll try to publish another release this week.
